src/building/bootstrapping.md

```diff
@@ -2,7 +2,83 @@
 
 This subchapter is about the bootstrapping process.
 
-When running `x.py` you will see output such as:
+## What is bootstrapping? How does it work?
+
+[Bootstrapping] is the process of using a compiler to compile itself.
+More accurately, it means using an older compiler to compile a newer version
+of the same compiler.
+
+This raises a chicken-and-egg paradox: where did the first compiler come from?
+It must have been written in a different language. In Rust's case it was
+[written in OCaml][ocaml-compiler]. However it was abandoned long ago and the
+only way to build a modern version of rustc is a slightly less modern
+version.
+
+This is exactly how `x.py` works: it downloads the current `beta` release of
+rustc, then uses it to compile the nightly compiler. The beta release is
```

Comment:
```suggestion
rustc, then uses it to compile the new compiler. The beta release is
```

Technically, most people don't compile "nightly" but whatever the master branch is...